### PR TITLE
CHI-1434: Create contact on new task.

### PR DESCRIPTION
**Pull request by stephenhand**
## Description

This PR makes 2 major changes:

* The contact is created in HRM either on accepting a twilio task or starting an offline contact, then updated with the full details later
* The redux store now holds all contacts under the 'existingContacts' store keyed by contactId, the old 'tasks' store keyed on taskSid previously used for unsaved contacts is now gone

To do in a follow up PR - align some of the changes in here with the refactor @acedeywin did concurrently to do contact updates via redux

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Related Issues
CHI-1433, CHI-1434

### Verification steps

* The correct version of HRM needs to be deployed before verifying this build, talk to @stephenhand 
* Regression test existing functionality
* Check that you get your forms back if you refresh partway through a new contact (except for the changes on the last tab)

### CODE REVIEW

**Comment by murilovmachado** [comment]
The refresh page scenario for **offline contacts** is not as "smooth" as the regular one, the user would need to click `+ Add Offline Contact` again, in order to see the contact he/she was filling in. 

I guess having that, for now, is a known compromise because it would need additional work to make that happen, right?

**Comment by stephenhand** [comment]
> The refresh page scenario for **offline contacts** is not as "smooth" as the regular one, the user needs to click `+ Add Offline Contact` again, in order to see the contact he/she was filling in.
> 
> I guess having that, for now, is a known compromise because it would need additional work to make that happen, right?

Yea - we should probably either blank the contact or add it to the left menu - probably easiest to do the latter because that's where we want to go eventually, I just wanted to avoid putting yet another change in this PR!

**Review by murilovmachado — APPROVED** [review]
Looks very good, Steve!

I can't say I looked at every line change, but overall this looks very good. Must've been a lot of work 😅 
Also, running locally, it works fine, too.

File: plugin-hrm-form/src/components/OfflineContact/AddOfflineContactButton.tsx [hunk 1]
```diff
@@ -34,16 +38,20 @@ type Props = OwnProps & ConnectedProps<typeof connector>;
...
 }) => {
   if (!currentDefinitionVersion) {
     return null;
   }
 
   const onClick = async () => {
-    recreateContactState(currentDefinitionVersion)(offlineContactTaskSid);
+    const { savedContact: newContact, metadata } = newContactState(currentDefinitionVersion)(false);
+    console.log('Onclick - creating contact');
+    const savedContact = await createContact(newContact, getHrmConfig().workerSid, getOfflineContactTaskSid());
```
[the remark this answers is not in the document]
Comment (stephenhand) [reply plugin-hrm-form/src/components/OfflineContact/AddOfflineContactButton.tsx:50]: Yes, I will address utilising the redux stuff Stephen did in a follow on PR

**Pull request merged**